[new-product] Add Veritas NetBackup

baggers27 opened the pull request:
This is a start to addressing #4215. It does not close that issue but this is the first pull request of the OS of the Netbackup appliances.  Separate pages may be needed for the Netbackup appliance hardware and software.

Information pulled from: https://www.veritas.com/support/en_US/eosl (under Appliances --> NetBackup). While the website uses start dates for extended support/sustaining support, I have treated these as the same as end of active support/extended support respectively (rather than maybe referring to the previous day).

I imagine new release data could potentially be automated by a custom python script.

baggers27 committed:
Create veritas-netbackup.md

baggers27 committed:
Update veritas-netbackup.md

baggers27 committed:
Update veritas-netbackup.md

baggers27 committed:
Update veritas-netbackup.md

baggers27 committed:
Update veritas-netbackup.md

baggers27 committed:
Update veritas-netbackup.md

usta reviewed (changes requested):
needs latest tags ( for each releasecycle ) , and releasecycle names couldnt have spaces or capitalletters

baggers27 committed:
Update veritas-netbackup.md

Set releaseColumn to false and fixed formatting on releaseCycle

baggers27 committed:
Update veritas-netbackup.md

Fix EOL vs EOS

baggers27 committed:
Update veritas-netbackup.md

Fix 3.1 and 3.1.1

baggers27 committed:
Update veritas-netbackup.md

Fix typo

usta reviewed (changes requested):
Product Validator: Invalid releases '5.1.1.mr5' for veritas-netbackup.md, expecting release (released on 2024-08-28) to be before 5.3 (released on 2023-11-27).
 Product Validator: Invalid releases '5.0.0.1.mr5' for veritas-netbackup.md, expecting release (released on 2023-11-08) to be before 5.1.1 (released on 2023-02-08).
 Product Validator: Invalid releases '4.1.0.1.mr5' for veritas-netbackup.md, expecting release (released on 2023-02-14) to be before 5.0 (released on 2022-04-25).
 Product Validator: Invalid releases '4.0.0.1.mr4' for veritas-netbackup.md, expecting release (released on 2022-06-13) to be before 4.1 (released on 2021-07-19).
 Product Validator: Invalid releases '3.3.0.2.mr2' for veritas-netbackup.md, expecting release (released on 2021-11-12) to be before 4.0 (released on 2021-03-01).
 
 Dont forget that releasecycles should be ordered with their releasedate not with their releasecyclename ( title )  so these 5 should be reordered with their releasedates  ( not with their number).
 
 Also It might look much better if you use `-` instead of `.` betweek mr parts 
 ( i mean for example instead of `5.3.0.1.mr4` -> `5.3.0.1-mr4` looks much better )

usta committed:
order release dates

Orders the releases by their date
Change .mr -> -mr style
Add spaces between cycles
Cosmatics changes

baggers27 committed:
Added description and support policy

baggers27 committed:
Add release labels and invert Booleans

baggers27 committed:
Fix typo

usta committed:
fix duplicates and reductants

baggers27 committed:
Fixed EOAS and EOL dates

usta reviewed (approved):
I'm not so familiar with the related product so i'm giving a ACT because of almost all of changes which i have suggested completed.
On the other hand , I suggest a second reviewer who is more knowledgable than me about this product

marcwrobel commented on products/veritas-netbackup.md:
```diff
@@ -0,0 +1,300 @@
+
+---
+
+> [Veritas NetBackup](https://www.veritas.com/protection/netbackup) is an enterprise-level data protection and backup solution
```
Is this about _Veritas NetBackup Appliance OS_ (used as the product name), or about _Veritas NetBackup_ ? [Those seem to be separate softwares](https://vox.veritas.com/discussions/netbackup/netbackup-version-and-appliance-os-version/904443) .

baggers27 replied:
For this PR, I am specifically referring to the appliance OS which is part of the NetBackup suite, and I can envisage follow-up PRs for the hardware and agent/server software.  I can add a line here that says "NetBackup appliances use the Veritas operating system (VxOS), which is a customized Linux operating system."  Maybe we should also change the name to VxOS but I wonder if that would be more confusing for people.  https://www.veritas.com/support/en_US/doc/96220900-127024912-0/v115997970-127024912

marcwrobel replied:
So Veritas NetBackup Appliance OS is in fact VxOS ? 

It seems to me the paragraph is about the NetBackup suite. To avoid confusion it would better to explicitly mention _Veritas NetBackup Appliance OS_ and not just _Veritas NetBackup_, and describe what it is.

marcwrobel commented on products/veritas-netbackup.md:
```diff
@@ -0,0 +1,300 @@
+
+> [Veritas NetBackup](https://www.veritas.com/protection/netbackup) is an enterprise-level data protection and backup solution
+
+Veritas offers full support from general availability for 3 to 4 years. At the end of full/active, support, extended support is available for 1-2 years with no new bug fixes, but access to existing patches and technical help. The final sustaining phase before EOSL lasts for 1 to 6 years with focus on severe service restoration or data retrieval issues.
```
On endoflife.date, products with no possibility of bug fixes are usually considered EOL. The reasoning behind that is that is a serious issue occurs and requires a fix, the only possibility you will have is upgrade to the next version.

So IMO it's OK to document extended support / sustaining support in the description, but they should not appear in the release table.

marcwrobel replied:
Looks like the description can be improved as it does not mention security fixes as seen on the following table on https://www.veritas.com/content/dam/support/terms/Veritas%20EOL%20Policy.pdf:
![image](https://github.com/user-attachments/assets/15fed2e8-b730-47eb-a4c2-b0d149260846)

Based on this table, there is:

- An active support phase (full support), with bug an security fixes.
- An security support phase (extended support), with only security fixes.

So this mean the sustaining phase must not appear in the table (but it's nice to document it in the description).

baggers27 replied:
I will rework and resubmit.

marcwrobel commented on products/veritas-netbackup.md:
```diff
@@ -0,0 +1,300 @@
+    - releaseCycle: "5.5.0.1"
+      releaseDate: 2025-02-12
+      eoas: true
+      eol: true
```
This means the latest version is already EOL, which cannot be true. I don't know what _Not applicable_ means, but I doubt it mean there is no support.

baggers27 replied:
I think based an another comment, if we removing the sustaining phases the eol will go back to being eol (rather than start of sustaining phase) and that would then be set to false (i.e still supported), with eoes (which is currently being used to represent eosl) disappearing.

marcwrobel reviewed (changes requested):
Thanks for this contribution @baggers27.

I see two main issue with this PR:

- I don't know the specific part of the product it is targeting. Veritas NetBackup seems to be a lot of things: hardware (appliances ?) as well as software (Veritas NetBackup Appliance OS, Media Server, OpsCenter...). And it seems each have its own support policy.
- I failed to understand what are the release cycles: are 5.3.0.1-mr3 and 5.3.0.1-mr4 in the same release cycle ? Why is there so many "Not applicable" mentions on https://sort.veritas.com/eosl ?

I am a bit confused about this product.

marcwrobel committed:
formatting

marcwrobel committed:
align filename, title and permalink

marcwrobel committed:
rework page

marcwrobel reviewed (approved):
Done.

I ended up documented only the Primary Support in the releases, because no new bug fixes are provided in extended or sustaining support.
